vm/src/obj/objtype.rs

```diff
@@ -462,19 +470,33 @@ fn take_next_base(mut bases: Vec<Vec<PyClassRef>>) -> Option<(PyClassRef, Vec<Ve
         }
     }
 
-    if let Some(head) = next {
-        for item in &mut bases {
-            if item[0].is(&head) {
-                item.remove(0);
-            }
+    let head = next?;
+    for item in &mut bases {
+        if item[0].is(&head) {
+            item.remove(0);
         }
-        return Some((head, bases));
     }
-    None
+    Some((head, bases))
 }
 
 fn linearise_mro(mut bases: Vec<Vec<PyClassRef>>) -> Option<Vec<PyClassRef>> {
```

Comment:
This should probably return `Result<Vec<PyClassRef>, String>` so the specific error can be propagated.